src/data_structure/binary_tree.rs

```diff
@@ -0,0 +1,66 @@
+#[derive(PartialEq)]
+pub struct Node<'a> {
+    val: &'a str,
+    left: Option<Box<Node<'a>>>,
+    right: Option<Box<Node<'a>>>,
+}
+
+impl<'a> Node<'a> {
+    pub fn insert(&mut self, new_val: &'a str) {
+        if self.val == new_val {
```

Comment:
Why are we not inserting if the incoming value matches the current?
I think you are trying to implement a `Binary Search Tree`